cmd/template_cmd.go

```diff
@@ -343,3 +367,114 @@ func TemplateServersList(c *cli.Context) error {
 	}
 	return nil
 }
+
+// =========== Template helpers =============
+
+// convertFlagParamsToCookbookVersions returns the json representation for the given friendly input format of cookbook versions assignation
+// i.e: "wordpress:0.1.0,nano=2.0.1,1password~>1.3.0"
+func convertFlagParamsToCookbookVersions(c *cli.Context, cbvsIn string) (map[string]interface{}, error) {
+	result := map[string]interface{}{}
+	cookbookVersions := make([]*types.CookbookVersion, 0)
+	for _, cbvIn := range strings.Split(cbvsIn, ",") {
+		values := regexp.MustCompile(`(.*?)(~>|=|>=|<=|>|<|:)(\w.*)`).FindStringSubmatch(cbvIn)
+		if len(values) == 0 {
+			return nil, fmt.Errorf("invalid input cookbook version format %s", cbvIn)
+		}
+		name, operator, version := values[1], values[2], values[3]
+		if _, found := result[name]; found {
+			return nil, fmt.Errorf("detected duplicated cookbook version name: %s", name)
+		}
+
+		// uploaded. It requires to map adequate version_id
+		if operator == ":" {
+			if len(cookbookVersions) == 0 {
+				// data is loaded only once
+				svc, formatter := WireUpCookbookVersion(c)
+				cbvs, err := svc.GetCookbookVersionList()
+				if err != nil {
+					formatter.PrintFatal("cannot receive uploaded cookbook versions data", err)
+				}
+				cookbookVersions = cbvs
+			}
+			for _, cbv := range cookbookVersions {
+				if name == cbv.Name && version == cbv.Version {
+					result[name] = map[string]interface{}{"version_id": cbv.ID}
+				}
+			}
+			// name does not match the available uploaded
+			if _, found := result[name]; !found {
+				return nil, fmt.Errorf("invalid uploaded cookbook version name: %s", name)
+			}
+		} else {
+			//supermarket
+			result[name] = map[string]interface{}{"version": operator + " " + version} // at any case, it should leave a space between operator and version
```

Comment:
You should validate version is 1 to 3 integers separated by dots. You could even add this to the `templateCookbookVersionValueRegexp` regexp since it can apply to uploaded cookbooks as well.

Reply:
Done